bundles/org.openhab.binding.enigma2/src/main/java/org/openhab/binding/enigma2/internal/Enigma2Client.java

```diff
@@ -203,6 +222,22 @@ private void processPowerResult(Document document) {
 
     private void processChannelResult(Document document) {
         channel = getString(document, "e2servicename");
+        // Add channel-Reference-ID if not known
+        if(!channels.containsKey(channel)) {
+            channels.put(channel, getString(document, "e2servicereference"));
+        }
+    }
+
+    private void processAnswerResult(Document document) {
+        if(asking) {
+            boolean state = getBoolean(document, "e2state");
+            if (state) {
+                String[] text = getString(document, "e2statetext").split(" ");
+                answer = text[text.length - 1].replace("!", "");
```

Comment:
What about an exception which can be thrown here due to `text` having an unexpected format?

Reply:
I think there couldn't occur an exception:
1) the result auf getString(document, "e2statetext") is never null
2) ther result of  split(" ") always returns an array with at least one item, also if there is no blank in the string, or the string has no characters
3) so text.length - 1 ist always greater or equals zero
4) the index of the text-array is always in range
5) the result of text[text.length - 1] is never null
6) replace("!", "") has no effect if there is no ! in the string

Reply:
thanks for the explanation, I wasn't sure about the "the result auf getString(document, "e2statetext") is never null".